<a href="https://colab.research.google.com/github/alearauj/EBA-Estatistica-do-Basico-ao-Avancado/blob/main/EBA_Projeto_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Projeto Final do curso EBA - Estatística do Básico ao Avançado**

####Preparando as liberações

In [1]:
# Importando todas as bibliotecas necessárias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Liberando acesso ao drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


####Preparando o dataset

In [3]:
# Importando o primeiro csv
df1 = pd.read_csv("//content/drive/MyDrive/Cursos/Renata Biaggi - Estatística do básico ao avançado/Projeto Final - Teste de Hipótese/DEMO_PHQ.csv", encoding = "UTF-8")

In [4]:
# Importando o segundo csv
df2 = pd.read_csv("/content/drive/MyDrive/Cursos/Renata Biaggi - Estatística do básico ao avançado/Projeto Final - Teste de Hipótese/PAG_HEI.csv", encoding = "UTF-8")

In [5]:
# Unindo os arquivos
df = pd.merge(df1, df2, on='SEQN')

In [6]:
#Checando se o tamanho do arquivo gerado está coerente com o esperado.
df.shape

(5334, 31)

####Explorando os dataframes


In [7]:
# Conhecendo as colunas
df.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  31130     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
1  31131     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  31132     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  31134     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4  31139     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0   

   DPQ090  ...  HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0     NaN  ...                   NaN                   NaN                NaN   
1     0.0  ...              0.000000              3.141062           4.071959   
2     0.0  ...              5.000000              4.448512           5.000000   
3     0.0  ...              2.217331              2.313900           5.000000   
4     0.0  ...              0.211544              0.000000           4.088849   

   HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                      NaN                  NaN                NaN   
1                 2.052399             0.684985           2.297100   
2                 0.000000             2.050769           7.369489   
3                 1.268648             2.958985           0.000000   
4                 4.221442            10.000000           2.845774   

   HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                      NaN              NaN                NaN   
1                 9.512964         0.438428           3.919226   
2                 8.887880         8.186877           9.099088   
3                 5.627488         5.863078           5.878686   
4                 0.000000        10.000000           9.658507   

   HEI2015_TOTAL_SCORE  
0                  NaN  
1            35.553899  
2            63.551696  
3            42.886826  
4            47.074393  

[5 rows x 31 columns]

In [8]:
# Conhecendo o dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5334 entries, 0 to 5333
Data columns (total 31 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   SEQN                      5334 non-null   int64  
 1   DPQ010                    4836 non-null   float64
 2   DPQ020                    4836 non-null   float64
 3   DPQ030                    4836 non-null   float64
 4   DPQ040                    4836 non-null   float64
 5   DPQ050                    4836 non-null   float64
 6   DPQ060                    4836 non-null   float64
 7   DPQ070                    4836 non-null   float64
 8   DPQ080                    4836 non-null   float64
 9   DPQ090                    4836 non-null   float64
 10  RIAGENDR                  5334 non-null   int64  
 11  RIDAGEYR                  5334 non-null   int64  
 12  RIDRETH1                  5334 non-null   int64  
 13  DMDEDUC                   5334 non-null   int64  
 14  INDFMINC

**Ponto de atenção**: Nem todas as colunas estão com o formato adequado para a informação que o dado quer apresentar. Por exemplo,todas as colunas contendo **DPQ**, **INDFMINC**  e **ADHERENCE**  estão como FLOAT quando deveriam ser INT64 já que possuem valores representados por códigos numéricos de valor inteiro. Além disso, a coluna **SEQN** está como INT64, o que não está errado, mas como essa coluna é nosso código identificador e não queremos fazer nenhum cálculo com ela, vamos convertê-la para STR.

In [19]:
#Encontrando valores nulos

df.isna().sum()

SEQN                          0
DPQ010                      498
DPQ020                      498
DPQ030                      498
DPQ040                      498
DPQ050                      498
DPQ060                      498
DPQ070                      498
DPQ080                      498
DPQ090                      498
RIAGENDR                      0
RIDAGEYR                      0
RIDRETH1                      0
DMDEDUC                       0
INDFMINC                     47
PAG_MINW                      0
ADHERENCE                     0
HEI2015C1_TOTALVEG          274
HEI2015C2_GREEN_AND_BEAN    274
HEI2015C3_TOTALFRUIT        274
HEI2015C4_WHOLEFRUIT        274
HEI2015C5_WHOLEGRAIN        274
HEI2015C6_TOTALDAIRY        274
HEI2015C7_TOTPROT           274
HEI2015C8_SEAPLANT_PROT     274
HEI2015C9_FATTYACID         274
HEI2015C10_SODIUM           274
HEI2015C11_REFINEDGRAIN     274
HEI2015C12_SFAT             274
HEI2015C13_ADDSUG           274
HEI2015_TOTAL_SCORE         274
dtype: i

Precisamos entender o que os valores nulos significam em cada uma das colunas para definirmos o que faremos com eles.

*   As colunas **DPQ** indicam o sentimento do respondente nas duas últimas semanas. As linhas vazias indicam que o respondente não respondeu sobre como ele estava se sentindo. Temos 498 respondentes que não preencheram tais informações. Isso representa quase 10% do dataframe.
*   A coluna **INDFMINC** indica a renda anual familiar em USD. Temos 47 respondentes que não informaram sua renda.
*   As colunas **HEI2015C** informam o hábito alimentar de cada respondente. Temos 274 respondentes que não responderam sobre seus hábitos.

In [ ]:
#somando os duplicados da base
df.duplicated().sum()

0

####Tratamento dos Dados

In [22]:
#Vamos manter as linhas vazias da coluna DPQ, mas vamos preenchê-la com o valor "7" que significa "Se recusou a responder".
DPQs_NaN = ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060', 'DPQ070', 'DPQ080', 'DPQ090']
df[DPQs_NaN] = df[DPQs_NaN].replace(np.nan,7)

In [27]:
#Vamos manter as linhas vazias da coluna INDFMINC, mas vamos preenchê-la com o valor "77" que significa "Se recusou a responder".

INDFMINC_NaN = ['INDFMINC']
df[INDFMINC_NaN] = df[INDFMINC_NaN].replace(np.nan,77)

In [30]:
#Vamos remover os respondentes que não possuem valores nas colunas de score de hábitos alimentares.

remover_vazio = ['HEI2015C1_TOTALVEG', 	'HEI2015C2_GREEN_AND_BEAN', 	'HEI2015C3_TOTALFRUIT', 	'HEI2015C4_WHOLEFRUIT', 	'HEI2015C5_WHOLEGRAIN', 	'HEI2015C6_TOTALDAIRY', 	'HEI2015C7_TOTPROT', 	'HEI2015C8_SEAPLANT_PROT', 	'HEI2015C9_FATTYACID', 	'HEI2015C10_SODIUM', 	'HEI2015C11_REFINEDGRAIN', 	'HEI2015C12_SFAT', 	'HEI2015C13_ADDSUG', 	'HEI2015_TOTAL_SCORE']
df = df.dropna(subset=remover_vazio)

In [24]:
#Convertendo a coluna INT para STR
coluna_int64 = 'SEQN'
df[coluna_int64] = df[coluna_int64].astype(str)

In [29]:
#Convertendo as colunas de FLOAT para INT
colunas_float = ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060', 'DPQ070', 'DPQ080', 'DPQ090','INDFMINC','ADHERENCE']
df[colunas_float] = df[colunas_float].astype('int64')

In [31]:
# Conhecendo o dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5060 entries, 1 to 5333
Data columns (total 31 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   SEQN                      5060 non-null   object 
 1   DPQ010                    5060 non-null   int64  
 2   DPQ020                    5060 non-null   int64  
 3   DPQ030                    5060 non-null   int64  
 4   DPQ040                    5060 non-null   int64  
 5   DPQ050                    5060 non-null   int64  
 6   DPQ060                    5060 non-null   int64  
 7   DPQ070                    5060 non-null   int64  
 8   DPQ080                    5060 non-null   int64  
 9   DPQ090                    5060 non-null   int64  
 10  RIAGENDR                  5060 non-null   int64  
 11  RIDAGEYR                  5060 non-null   int64  
 12  RIDRETH1                  5060 non-null   int64  
 13  DMDEDUC                   5060 non-null   int64  
 14  INDFMINC

In [32]:
df.describe()

DPQ010       DPQ020       DPQ030       DPQ040       DPQ050  \
count  5060.000000  5060.000000  5060.000000  5060.000000  5060.000000   
mean      0.733202     0.737352     0.957510     1.115810     0.743478   
std       1.790788     1.772935     1.811205     1.771272     1.780006   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     1.000000     0.000000   
75%       1.000000     1.000000     1.000000     1.000000     1.000000   
max       9.000000     9.000000     9.000000     9.000000     9.000000   

            DPQ060       DPQ070       DPQ080       DPQ090     RIAGENDR  ...  \
count  5060.000000  5060.000000  5060.000000  5060.000000  5060.000000  ...   
mean      0.640909     0.657905     0.584585     0.494071     1.520751  ...   
std       1.762569     1.759443     1.770863     1.728499     0.499619  ...   
min       0.000000     0.000000     0.000000     0.000000     1.000000  ...   
25%       0.000000     0.000000     0.000000     0.000000     1.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     2.000000  ...   
75%       0.000000     0.000000     0.000000     0.000000     2.000000  ...   
max       9.000000     9.000000     9.000000     9.000000     2.000000  ...   

       HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
count           5060.000000           5060.000000        5060.000000   
mean               1.764918              3.756967           4.434237   
std                1.915425              1.605226           1.008311   
min                0.000000              0.000000           0.000000   
25%                0.000000              2.591999           4.171156   
50%                1.026125              4.940948           5.000000   
75%                3.406820              5.000000           5.000000   
max                5.000000              5.000000           5.000000   

       HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
count              5060.000000          5060.000000        5060.000000   
mean                  2.593381             4.706986           4.762261   
std                   2.171836             3.237059           3.420252   
min                   0.000000             0.000000           0.000000   
25%                   0.000000             2.023710           1.587909   
50%                   2.588240             4.430397           4.833553   
75%                   5.000000             7.270784           7.650807   
max                   5.000000            10.000000          10.000000   

       HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
count              5060.000000      5060.000000        5060.000000   
mean                  5.798963         6.067985           6.220367   
std                   3.496261         3.178186           3.330305   
min                   0.000000         0.000000           0.000000   
25%                   2.925413         3.697080           3.724867   
50%                   6.234319         6.414811           6.864138   
75%                   9.253265         8.997076           9.454234   
max                  10.000000        10.000000          10.000000   

       HEI2015_TOTAL_SCORE  
count          5060.000000  
mean             49.971421  
std              12.324533  
min              16.059212  
25%              41.032289  
50%              49.512445  
75%              58.168013  
max              89.529383  

[8 rows x 30 columns]